### PR TITLE
Add batch request limit and response size limit in a RPC batch request

### PR DESCRIPTION
**Pull request by Kourin1996**
### Description

This PR adds limitations of number of batch items and response size in JSON-RPC request for avoiding an excessive memory usage.

After this change, a batch request can contain up to a maximum of 1000 requests, and the total size of all result data is limited to 25 MB.

The hardcoded boundary values are based on the default values in the current implementation of go-ethereum.

### Other changes

_Describe any minor or "drive-by" changes here._

### Tested

I added unit tests for each change

### Related issues

- Fixes #[issue number here]

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### CODE REVIEW

File: rpc/handler.go [hunk 1]
```diff
@@ -112,11 +124,21 @@ func (h *handler) handleBatch(msgs []*jsonrpcMessage) {
...
 	h.startCallProc(func(cp *callProc) {
 		answers := make([]*jsonrpcMessage, 0, len(msgs))
-		for _, msg := range calls {
+		for idx, msg := range calls {
 			if answer := h.handleCallMsg(cp, msg); answer != nil {
 				answers = append(answers, answer)
+				// Once total size of responses exceeds an allowed maximum,
+				// generate error messages for all remaining calls and stop further processing
+				if responseBytes += len(answer.Result); responseBytes > BatchResponseMaxSize {
+					for i := idx; i < len(calls); i++ {
```
Comment (karlb) [inline rpc/handler.go:136]: Since we added one `answer` above, don't we have to either
* start at `idx +1` to reject all the following requests (and live with the fact that we slightly exceeded `responseBytes`) or
* do this check before appending the current `answer` to `answers`?

File: rpc/handler.go [hunk 2]
```diff
@@ -112,11 +124,21 @@ func (h *handler) handleBatch(msgs []*jsonrpcMessage) {
...
+	responseBytes := 0
 	h.startCallProc(func(cp *callProc) {
 		answers := make([]*jsonrpcMessage, 0, len(msgs))
-		for _, msg := range calls {
+		for idx, msg := range calls {
 			if answer := h.handleCallMsg(cp, msg); answer != nil {
 				answers = append(answers, answer)
+				// Once total size of responses exceeds an allowed maximum,
+				// generate error messages for all remaining calls and stop further processing
+				if responseBytes += len(answer.Result); responseBytes > BatchResponseMaxSize {
```
Comment (gastonponti) [inline rpc/handler.go:135]: Technically `+=` is not thread safe.
I would recommend to use https://pkg.go.dev/sync/atomic#Int32.Add

File: rpc/handler.go (hunk 1 above)
Comment (gastonponti) [reply rpc/handler.go:136]: +1 to check before

File: rpc/handler.go (hunk 2 above)
Comment (gastonponti) [reply rpc/handler.go:135]: Nevermind, it's sequential on a thread not multi-thread

Comment (karlb) [reply rpc/handler.go:135]: `append` isn't thread-safe either. Is this an upstream bug or am I missing something that avoids problems here?

**Review by karlb — APPROVED** [review]
